lib/course_planner_web/controllers/offered_course_controller.ex

```diff
@@ -95,6 +126,46 @@ defmodule CoursePlannerWeb.OfferedCourseController do
     render(conn, "edit.html", offered_course: offered_course, changeset: changeset)
   end
 
+  def update(%{assigns: %{current_user: %{role: user_role, id: user_id}}} = conn,
+    %{"id" => id, "offered_course" => %{"syllabus" => syllabus}}) when user_role == "Teacher" do
+
+    offered_course =
+      OfferedCourse
+      |> Repo.get!(id)
+      |> Repo.preload([:term, :course, :students, :teachers, :classes])
+
+      can_user_update_this_offered_course? =
+        offered_course.teachers
+        |> Enum.any?(fn(teacher) -> teacher.id ==  user_id end)
+
+      if can_user_update_this_offered_course? do
+        {past_classes, next_classes} =
+          offered_course.classes
+          |> Classes.sort_by_starting_time()
+          |> Classes.split_past_and_next()
+
+        changeset = OfferedCourse.changeset(offered_course, %{syllabus: syllabus})
+
+        case Repo.update(changeset) do
+          {:ok, updated_offered_course} ->
```

Comment:
I believe you should send a notification here to the involved parties.

Reply:
hmmm, look I am not sure at all. that updating a description worth sending a notification ?
but maybe it does

Reply:
I guess so.